plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -210,14 +210,17 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
     override fun embedFunctionSignature(symbol: FirFunctionSymbol<*>): FunctionSignature {
         val retType = embedType(symbol.resolvedReturnTypeRef.type)
         val receiverType = symbol.receiverType
+        val isReceiverUnique = symbol.receiverParameter?.isUnique(session) ?: false
+        val isReceiverBorrowed = symbol.receiverParameter?.isBorrowed(session) ?: false
         return object : FunctionSignature {
             // TODO: figure out whether we want a symbol here and how to get it.
             override val receiver =
-                receiverType?.let { PlaceholderVariableEmbedding(ThisReceiverName, embedType(it)) }
+                receiverType?.let { PlaceholderVariableEmbedding(ThisReceiverName, embedType(it), isReceiverUnique, isReceiverBorrowed) }
             override val params = symbol.valueParameterSymbols.map {
-                FirVariableEmbedding(it.embedName(), embedType(it.resolvedReturnType), it)
+                FirVariableEmbedding(it.embedName(), embedType(it.resolvedReturnType), it, it.isUnique(session), it.isBorrowed(session))
             }
             override val returnType = retType
+            override val returnsUnique = symbol.isUnique(session) || symbol is FirConstructorSymbol
```

Comment:
Could you file an issue on space that our handling of constructors is not ideal (since we don't check whether the uniqueness is satisfied)?

Reply:
Are you thinking about cases like this one for the constructor not returning a unique? Or there is something more?
```kt
class Rec {
    val next: Rec = this
}
```

Reply:
Yeah, something like that.